DashAI/front/src/components/DatasetsTable.jsx

```diff
@@ -0,0 +1,205 @@
+import React, { useEffect } from "react";
+import PropTypes from "prop-types";
+
+import { DataGrid, GridActionsCellItem } from "@mui/x-data-grid";
+import DeleteIcon from "@mui/icons-material/Delete";
+import EditIcon from "@mui/icons-material/Edit";
+import AddIcon from "@mui/icons-material/Add";
+import {
+  Button,
+  Grid,
+  Paper,
+  Typography,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from "@mui/material";
+
+function EditDataset() {
```

Comment:
I would leave this component in another file, something like `EditDatasetModal`.